### ISSUE TITLE
Consider modal/dialog for Link Control popover when used

### ISSUE THREAD

**Issue by MaggieCabrera** [WordPress/gutenberg#51060]
While working on the component's tests, [we found](https://github.com/WordPress/gutenberg/pull/50996#discussion_r1208902343) that the popover doesn't have an accessible name. Also the focus is trapped inside the popover.

Should we be using a modal/dialog instead?

/cc @getdave

**Comment by getdave** [comment]
I actually don't know. I'd have to defer to folks with specific expertise in this area as it seems quite difficult to know what's best.

Do we have any components that fulfil the role? Surely our `Popover` is accessible and provides the correct `role` .etc?

**Comment by alexstine** [comment]
If you implement focus trap, it is probably best to implement modal while you are at it. Screen readers on Windows cannot navigate out of modals via browse/virtual mode but they can navigate away from popovers, non-modals. I define modals as the `<modal` tag or `role="modal"`. The Modal in wordpress/components should be fine.

**Comment by getdave** [comment]
Working on focus trap in https://github.com/WordPress/gutenberg/pull/54085
